### PR TITLE
Update docs and CI for py 3.13 support

### LINKED ISSUE #1342: Bump python support to 3.13 experimental
### 🚀 The feature

We should include builds for Python 3.13 though listed as experimental in order to start testing against FT/NoGIL python

### Motivation, pitch

NoGIL is going to make multi-threading a real possibility in Python, we should include builds for the next release of torchdata 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by andrewkho**
Enabling 3.13 CI for main/nightly, unsure if this needs more work for FTPython. Only enabled for Linux for now, as torch nightly does not yet have 3.13 for macos/windows

Fixes #1342 

### Changes
Update READMEs and CI to reflect 3.13 support
-
-

### CODE REVIEW

File: .github/workflows/stateful_dataloader_ci.yml [hunk 1]
```diff
@@ -29,6 +29,12 @@ jobs:
           - "3.10"
           - "3.11"
           - "3.12"
+          - "3.13"
+        exclude:
+          - os: macos-latest
+            python-version: "3.13"
+          - os: windows-latest
+            python-versino: "3.13"
```
Comment (divyanshk) [inline .github/workflows/stateful_dataloader_ci.yml:37]: Should be `python-version`.

File: .github/workflows/nodes_ci.yml [hunk 2]
```diff
@@ -29,6 +29,12 @@ jobs:
           - "3.10"
           - "3.11"
           - "3.12"
+          - "3.13"
+        exclude:
+          - os: macos-latest
+            python-version: "3.13"
+          - os: windows-latest
+            python-versino: "3.13"
```
Comment (divyanshk) [inline .github/workflows/nodes_ci.yml:37]: Should be `python-version`.

**Review by divyanshk — APPROVED** [review]
Minor comment to address, overall LGTM! Thank you..

File: .github/workflows/nodes_ci.yml (hunk 2 above)
Comment (andrewkho) [reply .github/workflows/nodes_ci.yml:37]: nice catch!

**Comment by andrewkho** [comment]
Test failure is unrelated, see https://github.com/pytorch/pytorch/issues/123649

**Pull request merged**